src/server/supplements/recorderSupplement.ts

```diff
@@ -402,18 +402,18 @@ export class RecorderSupplement {
     this._generator.signal(pageAlias, page.mainFrame(), { name: 'dialog', dialogAlias: String(++this._lastDialogOrdinal) });
   }
 
-  _captureSnapshot(sdkObject: SdkObject, metadata: CallMetadata, phase: 'before' | 'after' | 'action') {
+  async _captureSnapshot(sdkObject: SdkObject, metadata: CallMetadata, phase: 'before' | 'after' | 'action') {
     if (sdkObject.attribution.page) {
       const snapshotName = `${phase}@${metadata.id}`;
       this._snapshots.add(snapshotName);
-      this._snapshotter.captureSnapshot(sdkObject.attribution.page, snapshotName);
+      await this._snapshotter.captureSnapshot(sdkObject.attribution.page, snapshotName);
     }
   }
 
   async onBeforeCall(sdkObject: SdkObject, metadata: CallMetadata): Promise<void> {
     if (this._mode === 'recording')
       return;
-    this._captureSnapshot(sdkObject, metadata, 'before');
+    await this._captureSnapshot(sdkObject, metadata, 'before');
```

Comment:
It is important to no await here.

Reply:
hm, this should be commented then, maybe .catch too?